src/engine/sparqlExpressions/SparqlExpressionGenerators.h

```diff
@@ -157,6 +156,11 @@ auto applyOperation(size_t numElements, Operation&&, EvaluationContext* context,
   return std::apply(getResultFromValueGetters, ValueGetters{});
 }
 
-}  // namespace sparqlExpression::detail
+inline auto makeStringResultGetter(LocalVocab* localVocab) {
+  return [localVocab](const ad_utility::triple_component::LiteralOrIri& str) {
+    auto localVocabIndex = localVocab->getIndexAndAddIfNotContained(str);
+    return ValueId::makeFromLocalVocabIndex(localVocabIndex);
+  };
+}
```

Comment:
1. Give a short comment to this (`// return a lambda that takes a `LiteralOrIri` and converts it to an `Id` by adding it to the 
   local vocab.
4. As stated above, maybe we also want to abstract away the overloaded visit call for `IdOrLiteralOrIri`. (similar naming scheme, might be longer.